keras_cv/layers/preprocessing/random_zoom.py

```diff
@@ -0,0 +1,235 @@
+            self.height_lower = height_factor[0]
+            self.height_upper = height_factor[1]
+        else:
+            self.height_lower = -height_factor
+            self.height_upper = height_factor
+
+        if abs(self.height_lower) > 1.0 or abs(self.height_upper) > 1.0:
+            raise ValueError(
+                "`height_factor` must have values between [-1, 1], "
+                f"got {height_factor}"
+            )
+
+        self.width_factor = width_factor
+        if width_factor is not None:
+            if isinstance(width_factor, (tuple, list)):
+                self.width_lower = width_factor[0]
+                self.width_upper = width_factor[1]
+            else:
+                self.width_lower = -width_factor
+                self.width_upper = width_factor
+
+            if self.width_lower < -1.0 or self.width_upper < -1.0:
+                raise ValueError(
+                    "`width_factor` must have values larger than -1, "
+                    f"got {width_factor}"
+                )
+
+        preprocessing.check_fill_mode_and_interpolation(fill_mode, interpolation)
+
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.interpolation = interpolation
+        self.seed = seed
+
+    def get_random_transformation(self, image=None, **kwargs):
+        height_zoom = self._random_generator.random_uniform(
+            shape=[1, 1],
+            minval=1.0 + self.height_lower,
+            maxval=1.0 + self.height_upper,
+        )
+        if self.width_factor is not None:
+            width_zoom = self._random_generator.random_uniform(
+                shape=[1, 1],
+                minval=1.0 + self.width_lower,
+                maxval=1.0 + self.width_upper,
+            )
+        else:
+            width_zoom = height_zoom
+
+        return {"height_zoom": height_zoom, "width_zoom": width_zoom}
+
+    def augment_image(self, image, transformation, **kwargs):
+        image = preprocessing.ensure_tensor(image, self.compute_dtype)
+        original_shape = image.shape
+        image = tf.expand_dims(image, 0)
+        image_shape = tf.shape(image)
+        img_hd = tf.cast(image_shape[H_AXIS], tf.float32)
+        img_wd = tf.cast(image_shape[W_AXIS], tf.float32)
+        width_zoom = transformation["width_zoom"]
+        height_zoom = transformation["height_zoom"]
+        zooms = tf.cast(tf.concat([width_zoom, height_zoom], axis=1), dtype=tf.float32)
+        output = preprocessing.transform(
+            image,
+            self.get_zoom_matrix(zooms, img_hd, img_wd),
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            interpolation=self.interpolation,
+        )
+        output = tf.squeeze(output, 0)
+        output.set_shape(original_shape)
+        return output
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def compute_output_shape(self, input_shape):
```

Comment:
Don't think we need this

Reply:
Our BaseImageAugmentationLayer defines `_compute_output_signature` in lieu of the method `compute_output_signature` in base `keras.Layer`, and uses that instead, so in our KPLs we define `compute_image_signature` to define this, and we only need to do that if the output image shape is different from the input shape

Reply:
Got it! deleted compute_output_shape()